11.bulk_Morphology_Elastic_Network_LOCO/0.split_data/scripts/0.split_data_regression.py

```diff
@@ -0,0 +1,232 @@
+
+
+# In[6]:
+
+
+path = pathlib.Path(
+    f"../../../data/{cell_type}_preprocessed_sc_norm_aggregated_nomic.parquet"
+)
+
+data_df = pd.read_parquet(path)
+
+data_df.head()
+
+
+# In[7]:
+
+
+# subset each column that contains metadata
+metadata = data_df.filter(regex="Metadata")
+
+# get all columns that are not metadata except for metadata_Well
+data = data_df.drop(metadata.columns, axis=1)
+
+# get the metadata_Well column
+metadata_well = metadata[
+    ["Metadata_Well", "oneb_Metadata_Treatment_Dose_Inhibitor_Dose"]
+]
+
+data_df = pd.merge(data, metadata_well, left_index=True, right_index=True)
+
+
+# This model and code is both inspired and reused from: https://github.com/WayScience/phenotypic_profiling_model/blob/main/1.split_data/split_data.ipynb
+# The bulk of this work was done by **Roshan Kern** I have only made minor changes to the code to make it more modular and easier to use for my purposes.
+
+# In[8]:
+
+
+# variable test and train set splits
+# 100% test set
+# subset the following treatments for test set
+test_set_all = data_df[
+    data_df["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"].isin(test_100_percent)
+]
+# 75% test set and 25% train set
+test_set_75 = data_df[
+    data_df["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"].isin(test_75_percent)
+]
+
+test_100_and_75 = test_100_percent + test_75_percent
+
+# 50% test set and 50% train set
+# get all treatments that are not in the_test_set_all and the test_set_75
+test_set_50 = data_df[
+    ~data_df["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"].isin(test_100_and_75)
+]
+
+print(test_set_all.shape, test_set_75.shape, test_set_50.shape)
+
+
+# In[9]:
+
+
+# get the train test splits from each group
+# 100% test set
+test_set_all
+
+# 75% test set and 25% train set
+test_ratio = 0.75
+training_data_set_75, testing_data_set_75 = train_test_split(
+    test_set_75,
+    test_size=test_ratio,
+    stratify=test_set_75["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"],
+    random_state=0,
+)
+
+# 50% test set and 50% train set
+test_ratio = 0.5
+training_data_set_50, testing_data_set_50 = train_test_split(
+    test_set_50,
+    test_size=test_ratio,
+    stratify=test_set_50["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"],
+    random_state=0,
+)
+
+# verify that the correct splits have been made
+# 100% test set
+print(f"Shape for the 100% test set: {test_set_all.shape}\n")
+
+# 75% test set and 25% train set
+print(
+    f"Shape for the 75% test set: {training_data_set_75.shape};\nShape for the 75% train set: {testing_data_set_75.shape}\n"
+)
+
+# 50% test set and 50% train set
+print(
+    f"Shape for the 50% test set: {training_data_set_50.shape};\nShape for the 50% train set: {testing_data_set_50.shape}"
+)
+
+
+# In[10]:
+
+
+# combine all testing sets together while preserving the index
+testing_data_set = pd.concat(
+    [test_set_all, testing_data_set_75, testing_data_set_50], axis=0
+)
+testing_data_set = testing_data_set.sort_index()
+testing_data_set
```

Comment:
probably best to print just the `.head()`?

Reply:
removed print statements all together